src/scenes/scene-two.js

```diff
@@ -1,37 +1,235 @@
+        this.load.image('dialog-box', 'dialog-box.png');
         this.load.image('heart', 'heart.webp');
     }
 
     create() {
-        const town = this.add.image(0, -150, 'town').setOrigin(0, 0);
-        town.setScale(1.7);
-
-        this.add.image(150, 480, 'dialog-box').setOrigin(0).setScale(0.8).setDepth(2);
-
-        this.input.manager.enabled = true;
-
+        this.cameras.main.fadeIn(500, 0, 0, 0);
         for (let i = 0; i < heartSystem.lives; i++) {
             let heart = this.add.image(90 + i * 49, 60, 'heart').setScale(0.16).setDepth(2);
             heartSystem.hearts.push(heart);
         }
+
+        const town = this.add.image(0, -150, 'town').setOrigin(0, 0).setScale(1.7);
+
+        hero = this.add.image(-300, 150, 'hero').setOrigin(0, 0).setScale(1);
+
+        zacharius = this.add.image(1300, 205, 'zacharius').setOrigin(0, 0).setScale(1.3);
+        zacharius.flipX = true;
+
+        code = this.add.image(650, 250, 'code').setScale(0.4).setVisible(false);
+
+        const dialogBox = this.add.image(150, 480, 'dialog-box').setOrigin(0).setScale(0.8).setVisible(false);
+
+        sylvain = this.add.image(-300, 1000, 'sylvain').setOrigin(0, 0).setScale(1.1).setVisible(false);
+
+        this.moveIn(hero, 0);
+
+        dialog = this.add.text(220, 555, '', {
+            fontSize: '30px',
+            fontFamily: 'VT323',
+            wordWrap: {width: 850, useAdvancedWrap: true}
+            ,
+        });
+        name = this.add.text(265, 505, '', {
+            fontSize: '35px',
+            fontFamily: 'VT323',
+            wordWrap: {width: 850, useAdvancedWrap: true},
+        });
+
+        //long and contrived conditions game to get the right dialog
+        // at the right time and change scene
+        this.input.keyboard.on("keydown", () => {
+            if (step === 1) {
+                this.moveIn(zacharius, 900);
+                counter++
+                dialogBox.setVisible(true);
+                if (counter >= dialogFirstPart.length) {
+                    step = 2;
+                } else {
+                    this.continueDialog(dialogFirstPart);
+                }
+            }
+            if (step === 2) {
+                counter = -1;
+                this.optionDialog();
+                this.input.keyboard.on('keydown-A', () => {
+                    return [goodAnswer = true, step = 3]
+                })
+                this.input.keyboard.on('keydown-B', () => {
+                    return [goodAnswer = false, step = 3]
+                })
+            }
+            if (step === 3 && goodAnswer) {
+                option1.setVisible(false);
+                option2.setVisible(false);
+                counter++;
+                if (counter >= dialogRight.length) {
+                    isDialogOver = true;
+                } else {
+                    this.continueDialog(dialogRight);
+                }
+            }
+            if (step === 3 && goodAnswer === false) {
```

Comment:
```suggestion
            if (step === 3 && !goodAnswer) {
```
Jamais de triple egal false; c'est long et ça sert à rien

Reply:
c'est parce que je suis française, j'aime l'égalité.